lib/neo4j-core/query_clauses.rb

```diff
@@ -179,7 +179,7 @@ def attributes_string(attributes, prefix = '')
             if value.to_s.match(/^{.+}$/)
               "#{key}: #{value}"
             else
-              param_key = prefix + key.to_s
+              param_key = "#{prefix}#{key}".tap { |s| s.gsub!('::', '_') }
```

Comment:
The extra `tap`ing here kind of pains me.  In my speed tests, `gsub` is only 1.17x slower than `gsub!` in this case (methodology below).  Event `gsub` is doing 1.8 million iterations / second on my laptop.  Even if you're building a lot of queries doesn't seem like it's going to change anything noticeably.

```
2.2.0 :024 > s = "This::Is::A::Test"
 => "This::Is::A::Test"
2.2.0 :025 >
2.2.0 :026 >   Benchmark.ips do |x|
2.2.0 :027 >       x.config(:time => 5, :warmup => 2)
2.2.0 :028?>
2.2.0 :029 >       x.report("gsub") { s.gsub('::', '_') }
2.2.0 :030?>     x.report("gsub!") { s.gsub!('::', '_') }
2.2.0 :031?>
2.2.0 :032 >       x.compare!
2.2.0 :033?>   end
Calculating -------------------------------------
                gsub    37.986k i/100ms
               gsub!    83.872k i/100ms
-------------------------------------------------
                gsub      1.807M (±14.5%) i/s -      8.851M
               gsub!      2.111M (±12.6%) i/s -     10.400M

Comparison:
               gsub!:  2110668.7 i/s
                gsub:  1806592.5 i/s - 1.17x slower
```